### PR TITLE
Support Swift 5.0

### PR DESCRIPTION
**Pull request by chn-lyzhi**

### CODE REVIEW

File: APIKit.podspec [hunk 1]
```diff
@@ -9,7 +9,7 @@ Pod::Spec.new do |s|
   }
 
   s.ios.deployment_target = "8.0"
-  s.osx.deployment_target = "10.10"
+  s.osx.deployment_target = "10.11"
```
Comment (ikesyo) [inline APIKit.podspec]: Looks like you were hitting https://openradar.appspot.com/49262697 but this has been addressed in #270 (temporarily). So Please revert these deployment target changes.

File: APIKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -524,8 +516,10 @@
 				CLANG_ENABLE_MODULES = YES;
 				DEFINES_MODULE = YES;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.11;
 				SWIFT_OBJC_BRIDGING_HEADER = "";
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 5.0;
```
Comment (ikesyo) [inline APIKit.xcodeproj/project.pbxproj]: Please set this at the project level, not per-target level.

File: Package.swift [hunk 3]
```diff
@@ -7,7 +7,7 @@ let package = Package(
         .library(name: "APIKit", targets: ["APIKit"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/antitypical/Result.git", from: "4.0.0"),
+        .package(url: "https://github.com/antitypical/Result.git", from: "5.0.0"),
```
Comment (ikesyo) [inline Package.swift]: There is no Result 5.0.0 version. Why is this not removed?

File: README.md [hunk 4]
```diff
@@ -30,28 +30,30 @@ Session.send(request) { result in
...
 - Link your app with `APIKit.framework` and `Result.framework` in `Carthage/Build`.
 
 #### [CocoaPods](https://github.com/cocoapods/cocoapods)
 
-- Insert `pod 'APIKit', '~> 3.1'` to your Podfile.
+- Insert `pod 'APIKit', '~> 5.0'` to your Podfile.
 - Run `pod install`.
 
-Note: CocoaPods 1.1.0 is required to install APIKit 3.
+Note: CocoaPods 1.1.0 is required to install APIKit 5.
```
Comment (ikesyo) [inline README.md]: > CocoaPods 1.1.0

We use `swift_version` DSL, so this is not correct. Acutally we need CocoaPods 1.4.0 at least.

File: Package.swift (hunk 3 above)
Comment (chn-lyzhi) [reply Package.swift]: this is a mistake.

**Comment by chn-lyzhi** [comment]
Has fix.

**Comment by ikesyo** [comment]
Looks like you didn't successfully resolved the conflicts, please merge upstream's `master`  or rebase.

**Comment by chn-lyzhi** [comment]
Mabey don't need 'osx_image: xcode10.1, osx_image: xcode9.4, osx_image: xcode9.2' in .travis.yml file

**Pull request merged**